### PR TITLE
:scroll: Basic outline and template for LLM prac.

### PR DESCRIPTION
**Pull request by ruanvdmerwe**
Basic flow for the llm prac. 

The idea is to use A LOT from the prac last year to populate sections 1&2, then add in new content for section 3&4.

### CODE REVIEW

File: practicals/LLM_prac.ipynb [hunk 1]
```diff
@@ -0,0 +1,411 @@
+{
```
Comment (hereismari) [inline practicals/LLM_prac.ipynb:1]: <p>A few susgestions:</p><p> </p><ol><li>Given that most people attending are beginners can we have the Attention Mechanism section labeled as <code>Beginner</code> and target a very fundamental overview / breakdown of attention?</li><li>Change RLHF in content to "RLHF, Multimodal Transformers, and other relevant developments related to LLMs"</li><li>Rename: "Building a decoder only transformer from scratch" -&gt; "Understand and implement the fundamental building blocks of the Transformer Architecture"</li><li>Remove the "Understanding the RLHF training method" objective; we could instead say that "Introduce references to trending research and applications of LLMs" e.g. RLHF, MOE, Multimodal and multilingual transformers, sparsity and whatever else we could gather, for this section we can provide a very light definition/introduction and point to existing resources for more</li></ol>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deep-learning-indaba/indaba-pracs-2023/pull/6/discussion/#comment-1257321920'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

Comment (hereismari) [inline practicals/LLM_prac.ipynb:1]: <p>Maybe add a note mentioning that we def. want to cover this whole section during the practical, right? While section 2 and 3 might contain a lot of code/content so during the practical itself we might want to skip certain parts of it or focus on some specific sections</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deep-learning-indaba/indaba-pracs-2023/pull/6/discussion/#comment-1257321925'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='6'/>

Comment (hereismari) [inline practicals/LLM_prac.ipynb:1]: <p>typo: Transoformer -&gt; Transformer</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deep-learning-indaba/indaba-pracs-2023/pull/6/discussion/#comment-1257321932'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='10'/>

Comment (hereismari) [inline practicals/LLM_prac.ipynb:1]: <p>Consider update as suggested in the first comment</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deep-learning-indaba/indaba-pracs-2023/pull/6/discussion/#comment-1257321936'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='16'/>

Comment (EverlynAsiko) [inline practicals/LLM_prac.ipynb:1]: <p>typo: Tehory -&gt; Theory</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deep-learning-indaba/indaba-pracs-2023/pull/6/discussion/#comment-1259726399'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='7'/>

Comment (EverlynAsiko) [inline practicals/LLM_prac.ipynb:1]: <p>typo: Adapater -&gt; Adapter</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deep-learning-indaba/indaba-pracs-2023/pull/6/discussion/#comment-1259726402'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='14'/>

**Review by EverlynAsiko — APPROVED** [review]

**Pull request merged**